From 3aecfaa13a0a04cb9263d4eface931cd1dafe65a Mon Sep 17 00:00:00 2001
From: Hu Kejun <william.hu@rock-chips.com>
Date: Thu, 22 Nov 2018 12:43:50 +0800
Subject: [PATCH] media: rk-isp10: add TEST_PATTERN in s_ctrl

use V4L2_CID_TEST_PATTERN to enable test mode of sensor

Change-Id: I8a27d4ae072baecf62efb39d782bdfee897cc7bf
Signed-off-by: Hu Kejun <william.hu@rock-chips.com>
---
 drivers/media/platform/rk-isp10/cif_isp10.c                     | 1 +
 drivers/media/platform/rk-isp10/cif_isp10.h                     | 1 +
 drivers/media/platform/rk-isp10/cif_isp10_img_src_v4l2-subdev.c | 2 ++
 drivers/media/platform/rk-isp10/cif_isp10_v4l2.c                | 2 ++
 4 files changed, 6 insertions(+)

diff --git a/drivers/media/platform/rk-isp10/cif_isp10.c b/drivers/media/platform/rk-isp10/cif_isp10.c
index 2105c2088ee0..c7b508c7c9b6 100644
--- a/drivers/media/platform/rk-isp10/cif_isp10.c
+++ b/drivers/media/platform/rk-isp10/cif_isp10.c
@@ -6818,6 +6818,7 @@ int cif_isp10_s_ctrl(
 	case CIF_ISP10_CID_AUTO_FPS:
 	case CIF_ISP10_CID_HFLIP:
 	case CIF_ISP10_CID_VFLIP:
+	case CIF_ISP10_CID_TEST_PATTERN:
 		return cif_isp10_img_src_s_ctrl(dev->img_src,
 			id, val);
 	case CIF_ISP10_CID_FOCUS_ABSOLUTE:
diff --git a/drivers/media/platform/rk-isp10/cif_isp10.h b/drivers/media/platform/rk-isp10/cif_isp10.h
index 4b0398f9044e..3188c977ba93 100644
--- a/drivers/media/platform/rk-isp10/cif_isp10.h
+++ b/drivers/media/platform/rk-isp10/cif_isp10.h
@@ -162,6 +162,7 @@ enum cif_isp10_cid {
 	CIF_ISP10_CID_VBLANKING                   = 17,
 	CIF_ISP10_CID_ISO_SENSITIVITY             = 18,
 	CIF_ISP10_CID_MIN_BUFFER_FOR_CAPTURE      = 19,
+	CIF_ISP10_CID_TEST_PATTERN                = 20,
 
 };
 
diff --git a/drivers/media/platform/rk-isp10/cif_isp10_img_src_v4l2-subdev.c b/drivers/media/platform/rk-isp10/cif_isp10_img_src_v4l2-subdev.c
index ec27b510fa46..bb7e6214d89e 100644
--- a/drivers/media/platform/rk-isp10/cif_isp10_img_src_v4l2-subdev.c
+++ b/drivers/media/platform/rk-isp10/cif_isp10_img_src_v4l2-subdev.c
@@ -228,6 +228,8 @@ static int cif_isp10_v4l2_cid2v4l2_cid(u32 cif_isp10_cid)
 		return V4L2_CID_VFLIP;
 	case CIF_ISP10_CID_MIN_BUFFER_FOR_CAPTURE:
 		return V4L2_CID_MIN_BUFFERS_FOR_CAPTURE;
+	case CIF_ISP10_CID_TEST_PATTERN:
+		return V4L2_CID_TEST_PATTERN;
 	default:
 		cif_isp10_pltfrm_pr_err(NULL,
 			"unknown/unsupported CIF ISP20 ID %d\n",
diff --git a/drivers/media/platform/rk-isp10/cif_isp10_v4l2.c b/drivers/media/platform/rk-isp10/cif_isp10_v4l2.c
index 4a9265d0b54e..d4c3c5a35c34 100644
--- a/drivers/media/platform/rk-isp10/cif_isp10_v4l2.c
+++ b/drivers/media/platform/rk-isp10/cif_isp10_v4l2.c
@@ -284,6 +284,8 @@ static int cif_isp10_v4l2_cid2cif_isp10_cid(u32 v4l2_cid)
 		return CIF_ISP10_CID_AUTO_FPS;
 	case V4L2_CID_MIN_BUFFERS_FOR_CAPTURE:
 		return CIF_ISP10_CID_MIN_BUFFER_FOR_CAPTURE;
+	case V4L2_CID_TEST_PATTERN:
+		return CIF_ISP10_CID_TEST_PATTERN;
 	default:
 		cif_isp10_pltfrm_pr_err(NULL,
 			"unknown/unsupported V4L2 CID 0x%x\n",
-- 
2.35.3

